bin/kn

```diff
@@ -1,129 +1,132 @@
-  ansible             run Ansible. More help: kn ansible --help
-  ansible-playbook    run an Ansible playbook.
-                      More help: kn ansible-playbook --help
-  openstack           run a command against you Openstack provider: More help: kn openstack help
-  kubetoken           generate and print a kubeadm token
-  ssh <host>          ssh to <host> defined in ansible inventory file, default <host> is master
-  bash                run an interactive bash promt in the kubenow provisioners docker-image
-  az                  run Microsoft Azure CLI: More help: kn az help
-EOM
-
-# Validate command
-if [ "$#" -eq 0 ]; then
-  >&2 echo "Error: no arguments supplied"
-  echo "$USAGE"
-  exit 1
-fi
+  help                 print this message
+  init <CLOUD> <DIR>   creates a new project in the supplied directory
+                       CLOUD=<gce|aws|openstack|azure>
+  apply                apply configuration
+  destroy              destroy deployment
+  kubectl              run kubectl on the master. More help: kn kubectl help
+  helm                 run helm on the master. More help: kn helm help
+  terraform            run Terraform. More help: kn terraform --help
+  ansible              run Ansible. More help: kn ansible --help
+  ansible-playbook     run an Ansible playbook.
+                       More help: kn ansible-playbook --help
+  openstack            run a command against you Openstack provider: More help: kn openstack help
+  kubetoken            generate and print a kubeadm token
+  ssh <host>           ssh to <host> defined in ansible inventory file, default <host> is master
+  bash                 run an interactive bash promt in the kubenow provisioners docker-image
+  az                   run Microsoft Azure CLI: More help: kn az help
 
-COMMAND="$1"
-ALLOWED="help init terraform ansible ansible-playbook kubetoken apply destroy kubectl helm openstack az ssh bash"
-echo "$ALLOWED" | grep "\b$COMMAND\b" > /dev/null
-if [ "$?" = "1" ]; then
-  >&2 echo "Error: unrecognized command '$COMMAND'"
-  echo "$USAGE"
-  exit 1
-fi
-
-# Print help if required
-if [ "$COMMAND" = "help" ]; then
-  echo "$USAGE"
-  exit 0
-fi
+EOF
+}
 
 # If no version specified take stable
 if [ -z "$KUBENOW_VERSION" ] ; then
   KUBENOW_VERSION="latest"
 fi
+echo "Using KubeNow docker-image tag: $KUBENOW_VERSION"
 
-# Set directories to be mounted in docker
-USER_DIR="$PWD"
-USER_DIR_DOCKER="/var/userdir"
-# This is mainly for development where a local kubenow-dir will
-# be mounted on top of the kubenow dir in docker image
-if [ -n "$KUBENOW_DIR" ] ; then
-  OPTIONAL_KUBENOW_DIR_MNT="-v $KUBENOW_DIR:/opt/KubeNow"
-  echo "** Using local KubeNow directory: $KUBENOW_DIR **"
-fi
+# Get all local user group id:s (to be added to user inside docker-image)
+LOCAL_GROUP_IDS="$(id -G)"
 
 # Command name replacement
-first="$1"; shift; set -- "${first/ssh/ssh-kn}" "$@"
-first="$1"; shift; set -- "${first/init/init-kn}" "$@"
+if [ -n "$1" ]; then 
```

Comment:
Please assign $1 to a variable, or it is hard to read what it is about

Reply:
I did this for replacement block, but I don't think it should be done for the case block - since this is very much a standard way of accessing the bash command as "$1"